src/ScheduledJobs/Repositories/ParticipantRegistrationDynamoDbRepository.cs

```diff
@@ -1,37 +1,55 @@
+using System;
 using System.Collections.Generic;
+using System.Globalization;
+using System.Runtime.CompilerServices;
+using System.Threading;
 using System.Threading.Tasks;
 using Amazon.DynamoDBv2;
 using Amazon.DynamoDBv2.DataModel;
 using Dte.Common.Persistence;
 using ScheduledJobs.Contracts;
 using ScheduledJobs.Domain;
-using ScheduledJobs.Models;
 using ScheduledJobs.Settings;
 
 namespace ScheduledJobs.Repositories
 {
-    public class ParticipantRegistrationDynamoDbRepository : BaseDynamoDbRepository, IParticipantRegistrationDynamoDbRepository
+    public class ParticipantRegistrationDynamoDbRepository : BaseDynamoDbRepository,
+        IParticipantRegistrationDynamoDbRepository
     {
-        private readonly IAmazonDynamoDB _client;
         private readonly IDynamoDBContext _context;
         private readonly DynamoDBOperationConfig _config;
 
-        public ParticipantRegistrationDynamoDbRepository(IAmazonDynamoDB client, IDynamoDBContext context, AwsSettings awsSettings)
+        public ParticipantRegistrationDynamoDbRepository(IAmazonDynamoDB client, IDynamoDBContext context,
+            AwsSettings awsSettings)
             : base(client, context)
         {
-            _client = client;
             _context = context;
-            _config = new DynamoDBOperationConfig { OverrideTableName = awsSettings.ParticipantRegistrationDynamoDbTableName };
+            _config = new DynamoDBOperationConfig
+                { OverrideTableName = awsSettings.ParticipantRegistrationDynamoDbTableName };
         }
 
-        public async Task<IEnumerable<Participant>> GetAllAsync()
+        public async IAsyncEnumerable<Participant> GetAllAsync(
+            [EnumeratorCancellation] CancellationToken cancellationToken = default)
         {
-            return await _context.ScanAsync<Participant>(null, _config).GetRemainingAsync();
+            var search = _context.ScanAsync<Participant>(null, _config);
+
+            while (!search.IsDone)
+            {
+                var page = await search.GetNextSetAsync(cancellationToken);
+                foreach (var item in page)
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+                    yield return item;
+                }
+            }
         }
 
-        public async Task<Participant> GetParticipantAsync(string participantId)
+        public async Task<Participant> GetParticipantAsync(string participantId,
+            [EnumeratorCancellation] CancellationToken cancellationToken = default)
```

Comment:
The [EnumeratorCancellation] attribute only applies to IAsyncEnumerable return types. You should remove it here.